src/WebJobs.Script/Diagnostics/ApplicationInsightsMetricExporter.EventSource.cs

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics.Metrics;
+using System.Diagnostics.Tracing;
+
+namespace Microsoft.Azure.WebJobs.Script.Diagnostics
+{
+    public sealed partial class ApplicationInsightsMetricExporter
+    {
+        [EventSource(Name = $"{ScriptConstants.HostDiagnosticSourcePrefix}Diagnostics.{nameof(ApplicationInsightsMetricExporter)}")]
+        private sealed class Events : EventSource
```

Comment:
If we switch to `DiagnosticListener` instead of `EventSource`, the events will begin flowing to Kusto right away. I expect we’ll need these logs for troubleshooting, so having a configurable feature flag to enable them would be very helpful. 

Alternatively, we could add an intermediate listener that aggregates the events and then writes them through the `DiagnosticListener`. I’ve implemented a similar approach in `OpenTelemetryEventListener`.